File: app/internal/publisher/template/java/widgets/SettingsWidget.java
package com.websmith.sdk.widgets;

import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import java.io.File;
import java.io.FileReader;

public class SettingsWidget {
    public static void render() {
        System.out.println("\n--- SDK Settings ---\n");
        JsonObject config = loadConfig();
        if (config == null || config.entrySet().isEmpty()) {
            System.out.println("No configuration loaded.");
            return;
        }
        System.out.println("API Settings:");
        if (config.has("api")) {
            JsonObject api = config.getAsJsonObject("api");
            String url = api.has("url") ? api.get("url").getAsString() : "Not set";
            String version = api.has("version") ? api.get("version").getAsString() : "v1";
            int timeout = api.has("timeout") ? api.get("timeout").getAsInt() : 30000;
            int retryCount = api.has("retry_count") ? api.get("retry_count").getAsInt() : 3;
            String pubKey = api.has("public_key") ? api.get("public_key").getAsString() : "Not set";
            String maskedKey = pubKey.length() > 8 ? pubKey.substring(0, 8) + "..." : pubKey;
            System.out.println("  URL: " + url);
            System.out.println("  Version: " + version);
            System.out.println("  Timeout: " + timeout + "ms");
            System.out.println("  Retry Count: " + retryCount);
            System.out.println("  Public Key: " + maskedKey);
        }
        if (config.has("product")) {
            JsonObject product = config.getAsJsonObject("product");
            System.out.println("\nProduct Settings:");
            System.out.println("  ID: " + (product.has("id") ? product.get("id").getAsString() : "Not set"));
            System.out.println("  Name: " + (product.has("name") ? product.get("name").getAsString() : "Not set"));
        }
        if (config.has("trial")) {
            JsonObject trial = config.getAsJsonObject("trial");
            System.out.println("\nTrial Settings:");
            System.out.println("  Enabled: " + (trial.has("enabled") && trial.get("enabled").getAsBoolean() ? "Yes" : "No"));
            System.out.println("  Days: " + (trial.has("days") ? trial.get("days").getAsInt() : 0));
        }
        if (config.has("offline")) {
            JsonObject offline = config.getAsJsonObject("offline");
            System.out.println("\nOffline Settings:");
            System.out.println("  Cache Days: " + (offline.has("cache_days") ? offline.get("cache_days").getAsInt() : 0));
        }
        if (config.has("branding")) {
            JsonObject branding = config.getAsJsonObject("branding");
            System.out.println("\nBranding:");
            System.out.println("  Company: " + (branding.has("company_name") ? branding.get("company_name").getAsString() : "Not set"));
            System.out.println("  Support Email: " + (branding.has("support_email") ? branding.get("support_email").getAsString() : "support@websmithdigital.com"));
        }
    }

    private static JsonObject loadConfig() {
        String[] paths = {"config/api-config.json", "config/api-config.json"};
        for (String path : paths) {
            try {
                File f = new File(path);
                if (f.exists()) {
                    try (FileReader reader = new FileReader(f)) {
                        return JsonParser.parseReader(reader).getAsJsonObject();
                    }
                }
            } catch (Exception ignored) {}
        }
        return null;
    }
}

File: app/internal/publisher/template/dotnet/widgets/SettingsWidget.cs
using System;
using System.IO;
using System.Text.Json;

namespace WebsmithSDK.Widgets;

public static class SettingsWidget
{
    public static void Render()
    {
        Console.WriteLine("\n--- SDK Settings ---\n");
        var config = LoadConfig();
        if (config == null)
        {
            Console.WriteLine("No configuration loaded.");
            return;
        }
        var root = config.RootElement;
        Console.WriteLine("API Settings:");
        if (root.TryGetProperty("api", out var api))
        {
            Console.WriteLine($"  URL: {GetString(api, "url", "Not set")}");
            Console.WriteLine($"  Version: {GetString(api, "version", "v1")}");
            Console.WriteLine($"  Timeout: {GetInt(api, "timeout", 30000)}ms");
            Console.WriteLine($"  Retry Count: {GetInt(api, "retry_count", 3)}");
            var pubKey = GetString(api, "public_key", "Not set");
            Console.WriteLine($"  Public Key: {(pubKey.Length > 8 ? pubKey[..8] + "..." : pubKey)}");
        }
        if (root.TryGetProperty("product", out var product))
        {
            Console.WriteLine("\nProduct Settings:");
            Console.WriteLine($"  ID: {GetString(product, "id", "Not set")}");
            Console.WriteLine($"  Name: {GetString(product, "name", "Not set")}");
        }
        if (root.TryGetProperty("trial", out var trial))
        {
            Console.WriteLine("\nTrial Settings:");
            Console.WriteLine($"  Enabled: {(GetBool(trial, "enabled") ? "Yes" : "No")}");
            Console.WriteLine($"  Days: {GetInt(trial, "days", 0)}");
        }
        if (root.TryGetProperty("offline", out var offline))
        {
            Console.WriteLine("\nOffline Settings:");
            Console.WriteLine($"  Cache Days: {GetInt(offline, "cache_days", 0)}");
        }
        if (root.TryGetProperty("branding", out var branding))
        {
            Console.WriteLine("\nBranding:");
            Console.WriteLine($"  Company: {GetString(branding, "company_name", "Not set")}");
            Console.WriteLine($"  Support Email: {GetString(branding, "support_email", "support@websmithdigital.com")}");
        }
    }

    private static string GetString(JsonElement obj, string key, string def)
    {
        return obj.TryGetProperty(key, out var p) && p.ValueKind == JsonValueKind.String
            ? p.GetString() ?? def : def;
    }

    private static int GetInt(JsonElement obj, string key, int def)
    {
        return obj.TryGetProperty(key, out var p) ? p.GetInt32() : def;
    }

    private static bool GetBool(JsonElement obj, string key)
    {
        return obj.TryGetProperty(key, out var p) && p.GetBoolean();
    }

    private static JsonDocument? LoadConfig()
    {
        var paths = new[] { "config/api-config.json", "config/api-config.json" };
        foreach (var path in paths)
        {
            if (File.Exists(path))
            {
                try { return JsonDocument.Parse(File.ReadAllText(path)); }
                catch { }
            }
        }
        return null;
    }
}
